lang/syn/src/parser/context.rs

```diff
@@ -40,6 +40,43 @@ impl CrateContext {
             modules: ParsedModule::parse_recursive(root.as_ref())?,
         })
     }
+
+    // Perform Anchor safety checks on the parsed create
+    pub fn safety_checks(&self) -> Result<(), anyhow::Error> {
+        // Check all structs for unsafe field types, i.e. AccountInfo and UncheckedAccount.
+        for (_, ctx) in self.modules.iter() {
+            for unsafe_field in ctx.unsafe_struct_fields() {
+                // Check if unsafe field type has been documented with a /// SAFETY: doc string.
+                let is_documented_or_safe = unsafe_field.attrs.iter().any(|attr| {
+                    attr.tokens.clone().into_iter().any(|token| match token {
+                        // Check for comments containing SAFETY: or CHECK:
+                        proc_macro2::TokenTree::Literal(s) => {
+                            s.to_string().contains("SAFETY:") || s.to_string().contains("CHECK:")
+                        }
+                        _ => false,
+                    })
+                });
+                if !is_documented_or_safe {
+                    let ident = unsafe_field.ident.as_ref().unwrap();
+                    let span = ident.span();
+                    // Error if undocumented.
+                    return Err(anyhow!(
+                        r#"
+        {}:{}:{}
+        Struct field "{}" is unsafe, but is not documented.
+        Please add a `/// SAFETY:` doc comment to the field enumerating potential security risks.
```

Comment:
> enumerating potential security risks.

IIUC it's meant to explain why the unchecked type is safe to use. So imo something like
```
Please add a `/// SAFETY:` doc comment to the field explaining why no checks are necessary.
```
would be better

Reply:
yea true. then maybe 

```
explaining why no checks through types are necessary
```

Reply:
Yea that sounds nicer 👍